client/landing/stepper/declarative-flow/site-migration-flow.ts

```diff
@@ -185,6 +185,10 @@ const siteMigration: Flow = {
 						action: SiteMigrationIdentifyAction;
 					};
 
+					saveSiteSettings( siteSlug, {
```

Comment:
I think this should have a safety check on `siteSlug` and should also have an await. 

Here's how we do it in other places: https://github.com/Automattic/wp-calypso/blob/d7ed593824d3fd2e56088d50df3f5005f87ca797/client/landing/stepper/declarative-flow/site-migration-flow.ts#L264